client/src/main/java/io/pravega/client/admin/ReaderGroupManager.java

```diff
@@ -73,11 +72,10 @@ public static ReaderGroupManager withScope(String scope, ClientConfig clientConf
      * may block.
      * @param groupName The name of the group to be created.
      * @param config The configuration for the new ReaderGroup.
-     * @return True if ReaderGroup was created.
-     * @throws ConfigMismatchException If the reader group already exists with a different configuration. Use {@link ReaderGroup#resetReaderGroup} to change
-     * the reader group configuration.
+     * @return The {@link ReaderGroupConfig} of the ReaderGroup. If a ReaderGroup already exists then the older configuration
+     * is returned.
      */
-    boolean createReaderGroup(String groupName, ReaderGroupConfig config) throws ConfigMismatchException;
+    ReaderGroupConfig createReaderGroup(String groupName, ReaderGroupConfig config);
```

Comment:
It would help to change the `createReaderGroup` API to a `getOrCreate()` API, so that it would create the ReaderGroup if one does not exist or else just return a reference to a Reader Group object. 
With a `getOrCreateReaderGroup()` API users would not need to issue 2 separate calls `createReaderGroup` and then again `getReaderGroup`.

However, I also see is that the Reader Group class does not hold a reference to the Reader Group config, which also needs to be fixed..

Reply:
A `ReaderGroup` object uses a StateSynchronizer. This implies the user would have to close this ReaderGroup object which would be returned as part of a `getOrCreate()`, which I think would be cumbersome for users. 
The ReaderGroupManager already has an API for fetching the ReaderGroup object.
```java
    /**
     * Returns the requested reader group.
     * 
     * @param groupName The name of the group
     * @return Reader group with the given name
     * @throws ReaderGroupNotFoundException If the reader group does not exist.
     */
    ReaderGroup getReaderGroup(String groupName) throws ReaderGroupNotFoundException;
```